### PR TITLE
Run benchmarks as unit tests on CI

### LINKED ISSUE #60: test_bench.py: Snappy tests fail
Something I've stumbled upon packaging this for nixpkgs https://github.com/NixOS/nixpkgs/pull/124862.
Here is the build log https://gist.github.com/veprbl/345d882b01923f31fcfc75c1238d305b
Fastparquet's round trip tests seem to pass just fine. Perhaps they don't use this code path?

### PR DESCRIPTION
**Pull request by messense**
Fixes #60

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,6 +3,9 @@ BASE_BENCH_CMD = python -m pytest -v --benchmark-sort name --benchmark-only benc
 test:
 	python -m pytest tests -v --ignore benchmarks
 
+test-bench:
+	python -m pytest -v --benchmark-disable benchmarks/
```
Comment (messense) [inline Makefile:7]:
```
--benchmark-disable   Disable benchmarks. Benchmarked functions are only ran once and no stats are reported. Use this is you want to run the test but don't do any benchmarking.
```

File: benchmark-requirements.txt [hunk 2]
```diff
@@ -3,3 +3,5 @@ python-snappy==0.5.4
 lz4==3.1.0
 brotlipy==0.7.0
 zstd==1.4.5.1
+numpy
+memory-profiler
```
Comment (milesgranger) [inline benchmark-requirements.txt:7]: Nice, thanks!

File: benchmarks/test_bench.py [hunk 3]
```diff
@@ -110,7 +110,8 @@ def test_cramjam_snappy_de_compress_into(benchmark, op, file):
 
     operation = getattr(snappy, op)
     buffer = np.zeros(
-        len(data) if op == "decompress_into" else len(compressed_data), dtype=np.uint8
+        len(data) if op == "decompress_into" else len(compressed_data),
+        dtype=np.uint8,
```
Comment (milesgranger) [inline benchmarks/test_bench.py:114]: ah yes, this would have never evaluated to true. :facepalm:

**Review by milesgranger — APPROVED** [review]
Goodie! Thanks for getting at this so quickly, really appreciate your help! :+1:

**Comment by messense** [comment]
Saw a warning in test benchmarks step, I wonder if it's related to pyo3? cc @davidhewitt

```
=============================== warnings summary ===============================
benchmarks/test_bench.py: 28 warnings
  /home/runner/work/pyrus-cramjam/pyrus-cramjam/benchmarks/test_bench.py:41: DeprecationWarning: PY_SSIZE_T_CLEAN will be required for '#' formats
    return decompress(compress(data, **kwargs))

-- Docs: https://docs.pytest.org/en/stable/warnings.html
====================== 220 passed, 28 warnings in 30.37s =======================
```

**Comment by davidhewitt** [comment]
I don't think it's related to PyO3, we don't use the `PyArg_Parse` family of functions in PyO3.

My guess is one of the other compression libs with a C implementation is responsible. It'd probably be possible to work out which one and report an issue to that project if you're feeling charitable 😄

**Pull request merged**